x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -492,9 +497,12 @@ protected Clock getClock() {
         );
         this.tokenService.set(tokenService);
         components.add(tokenService);
-
+        if (BOOTSTRAP_ELASTIC_PASSWORD.exists(settings) == false) {
+            securityIndex.get().addStateListener(this::generateElasticPassword);
+        }
```

Comment:
If this is a couple of lines lower, then we can remove the `SetOnce` entirely and just pass the `NativeUsersStore` to the listener object's constructor (if we replace the method reference with a class).